include/language-factory.php

```diff
@@ -98,6 +98,31 @@ public static function get_from_terms( array $terms ) {
 		$flag_props = self::get_flag( $data['flag_code'], $data['name'], $data['slug'], $data['locale'] );
 		$data       = array_merge( $data, $flag_props );
 
+		$static_pages = array(
+			'page_on_front'  => 0,
+			'page_for_posts' => 0,
+		);
+
+		/**
+		 * Filters the page_on_front and page_for_posts options before the language is created
+		 * and its home and search URLs are initialized.
+		 *
+		 * @since 3.4
+		 *
+		 * @param array  $static_pages {
+		 *     Array of language options.
+		 *
+		 *     @type string $page_on_front  Static page id of the front page in this language.
+		 *     @type string $page_for_posts Static page id of the posts page in this language.
+		 * }
+		 * @param array $data Language data.
+		 *
+		 * @phpstan-param array{page_on_front: int, page_for_posts: string} $static_pages
+		 * @phpstan-param non-empty-array<string, mixed> $data
+		 */
+		$_static_pages = apply_filters( 'pll_static_pages', $static_pages, $data );
```

Comment:
I propose to use a more generic filter common with #1195.

Reply:
Done in d7a53cc35ac9e50030dbc56dc127071451c9fb34